app/controllers/add-preprint.js

```diff
@@ -11,6 +12,13 @@ export default Ember.Controller.extend({
         uploadMultiple: false,
         method: 'PUT'
     },
+    canEdit: Ember.computed('isAdmin', 'isRegistration', function() {
+        return this.get('isAdmin') && !(this.get('model').get('registration'));
+    }),
+    isAdmin: Ember.computed(function() {
+        return this.get('model').get('currentUserPermissions').indexOf('admin') >= 0;
```

Comment:
Small syntax note- you can use `.get('model.currentUserPermissions')` to fetch nested values in a single get.